staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/controller/controller.go

```diff
@@ -0,0 +1,291 @@
+
+	// add dummy item back to the queue to trigger file content processing.
+	d.queue.AddRateLimited(key)
+
+	return true
+}
+
+// loadEncryptionConfig processes the next set of content from the file.
+func (d *DynamicKMSEncryptionConfigContent) loadEncryptionConfig() error {
+	encryptionConfig, err := d.readEncryptionFileContent(d.filePath)
+	if err != nil {
+		return err
+	}
+	if len(encryptionConfig) == 0 {
+		return fmt.Errorf("encryption config file is empty %q", d.name)
+	}
+
+	// compute hash for the encryption config file content.
+	encryptionConfigHash, err := d.computeHash(encryptionConfig)
+	if err != nil {
+		klog.Warningf("failed to compute hash for encryption config file %s: %v", d.filePath, err)
+		return err
+	}
+
+	// check if encryptionConfig is different from the current. Do nothing if they are the same.
+	if !d.hasEncryptionConfigChanged(encryptionConfigHash) {
+		return nil
+	}
+
+	err = d.processEncryptionConfig()
+	if err != nil {
+		return err
+	}
+
+	// update local copy of recent config content once processing is successful.
+	d.encryptionConfig.Store(encryptionConfigHash)
+	klog.V(2).InfoS("Loaded new kms encryption config content", "name", d.name)
+
+	return nil
+}
+
+func (d *DynamicKMSEncryptionConfigContent) readEncryptionFileContent(filePath string) ([]byte, error) {
+	return os.ReadFile(d.filePath)
+}
+
+// processEncryptionConfig processes the encryption config.
+func (d *DynamicKMSEncryptionConfigContent) processEncryptionConfig() error {
+	// get context to close the old transformers.
+	ctx, closeTransformers := wait.ContextForChannel(d.apiServerDrainCh)
+
+	transformerOverrides, _, err := encryptionconfig.LoadEncryptionConfig(d.filePath, ctx.Done())
+	if err != nil {
+		klog.Warning("KMS encryption configuration is not valid.", err)
+		return err
+	}
+
+	// update transformers.
+	// new inFlight variable to track transformers that will be set in here.
+	var inFlight atomic.Int32
+	cancel, oldInFlight := d.dynamicTransformers.Set(transformerOverrides, closeTransformers, &inFlight)
+
+	if oldInFlight != nil && cancel != nil {
+		// poll until oldInFlight is zero.
+		go wait.PollImmediateUntil(1*time.Second, func() (bool, error) {
+			// check for oldInFlight to be zero to ensure they are not in use.
+			if oldInFlight.Load() == 0 {
+				cancel()
+				return true, nil
+			}
+			return false, nil
+		}, ctx.Done())
+	}
+
+	return nil
+}
+
+// hasEncryptionConfigChanged returns true if it has changed
+func (d *DynamicKMSEncryptionConfigContent) hasEncryptionConfigChanged(newEncryptionConfigHash string) bool {
+	if d.encryptionConfig.Load().(string) == newEncryptionConfigHash {
+		klog.V(4).InfoS("Encryption config has not changed", "name", d.name)
+		return false
+	}
+
+	return true
+}
+
+func (d *DynamicKMSEncryptionConfigContent) computeHash(data []byte) (string, error) {
+	hash := sha256.New()
+	_, err := hash.Write(data)
+	if err != nil {
+		return "", err
+	}
```

Comment:
```suggestion
	hash := sha256.Sum256(data)
```

And drop the error return.